### DISCUSSION TITLE
Translating books

### DISCUSSION THREAD

**Discussion by Kentoseth** [WeblateOrg/weblate#5437] [Q&A] [answered]
Hello,

This is a 2-part discussion.

 1. It is possible to use Weblate to translate longer texts like books? Will the translation-memory be able to "learn from" longer strings like complex sentences?
 2. If it is possible to use Weblate for books, what is the best format/structure for the text? It doesn't look like .txt files are supported for import for monolingual files
 3. Based on 2 above, what format will be suitable for importing existing translations for the CAT tool to "learn from"?

Looking forward to your reply.

**Comment by nijel** [comment, marked as answer]
I have no experience myself, but there is [The Debian Administrator's Handbook](https://debian-handbook.info/) translated using Weblate, so you might want to look at how they do that: https://hosted.weblate.org/projects/debian-handbook/. It uses [Publican](https://sourceware.org/publican/) to build and localize.

**Comment by harcesz** [comment]
Looking for the same. There seems to be a massive gap in the open source translation tools niche; tools for collectively translating longer texts like articles and books. I do a lot of these, and pretty much the only option is setting up omegat and with a group of not very tech savvy translators it's a full time job in itself, and doesn't allow easy way to join a collaboration project. http://zanata.org/ seemed to be close to optimal for that, but it's not developed since a few years and tricky to install, so maybe that could be an option or extension for weblate?
The thing is; creating repositories puts a massive technical overhead for non-techie users, and setting a text to be in the right format in a repo is just more wasted time from this perspective. So people turn to commercial, or exploitive (as in using their work) tools. 

A dream solution here would be to be able to just upload a text to weblate, for it to be displayed for translation - possibly each sentence treated as a string, or exposing options for users to decide how to cut it. The other key thing would be a view with a stronger emphasis on context (seeing few sentences before and after the current one).

**Reply by Kentoseth** [reply]
Have you seen the Translation management systems (TMS) available here: [18 open source translation tools](https://opensource.com/article/17/6/open-source-localization-tools) ?

The list they mention is:

    Pontoon
    Pootle
    Weblate
    Translate5
    GlobalSight
    Zanata
    Jabylon

I was going to settle on OmegaT but Pontoon, Pootle and Translate5 all look like they might serve the needs of translating longer texts.

**Reply by nijel** [reply]
You can upload document to Weblate and translate it (look for "Translate document" in https://docs.weblate.org/en/latest/admin/projects.html#adding-translation-projects-and-components). The file formats support is limited though and the implementation has some limitations, see https://github.com/WeblateOrg/weblate/discussions/6376.

**Reply by comradekingu** [reply]
https://hosted.weblate.org/languages/nb_NO/madewithcc/#information is a book translated on Weblate.

**Reply by Kentoseth** [reply]
@comradekingu 

Were you able to take advantage of Translation Memories(known as TMs by translators) after translating a few sentences in Weblate?

**Reply by comradekingu** [reply]
I don't really use it when translating.
I think it be set up as imported from the general pool, from a specific resource, shared within your project, or available as history — with or without marking it as needing work.
If you are translating free-form text it is going to be next to useless.
